lib/interfaces/storage.js

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2018 Scott Bender <scott@scottbender.net>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const _ = require('lodash')
+const debug = require('debug')('signalk-server:interfaces:storage')
+const fs = require('fs')
+const path = require('path')
+const jsonpatch = require('json-patch')
+
+const storageUrls = [
+  '/signalk/v1/storage/:appid/:version/*',
+  '/signalk/v1/storage/:appid/:version'
+]
+
+const userStorageUrls = [
+  '/signalk/v1/storage/user/:appid/:version/*',
```

Comment:
Possible appid and username clash.

Reply:
I changed the url for global to include 'global'